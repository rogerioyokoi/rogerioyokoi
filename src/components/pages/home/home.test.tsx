import { useMediaQuery } from '@react-hook/media-query';
import { render, screen } from '@testing-library/react';
import { Mock, beforeEach, describe, expect, it, vi } from 'vitest';
import { axe } from 'vitest-axe';
import { baseDescription } from '../../../assets/data/meta.const';
import Home from './home';

// Mock para o hook useMediaQuery
vi.mock('@react-hook/media-query', () => ({
  useMediaQuery: vi.fn(),
}));

describe('Components > Pages > Home', () => {
  beforeEach(() => {
    // Define o comportamento padrão do hook useMediaQuery
    (useMediaQuery as Mock).mockReturnValue(false);
  });

  describe('Snapshot', () => {
    it('Should match snapshot', () => {
      const { container } = render(<Home />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('Accessibility Tests (A11y)', () => {
    it('Should not have accessibility violations', async () => {
      const { container } = render(<Home />);
      expect(await axe(container)).toHaveNoViolations();
    });
  });

  describe('Rendering', () => {
    it('Should render Avatar component correctly', () => {
      render(<Home />);
      const avatar = screen.getByAltText('Foto de rosto do proprietário do website: Rogério Yokoi');
      expect(avatar).toBeInTheDocument();
    });

    it('Should render Title component with correct text', () => {
      render(<Home />);
      const title = screen.getByText('Olá, eu sou Rogério Yokoi');
      const subTitle = screen.getByText('Engenheiro de software Front-End');
      expect(title).toBeInTheDocument();
      expect(subTitle).toBeInTheDocument();
    });

    it('Should render the description correctly', () => {
      render(<Home />);
      const description = screen.getByText(baseDescription);
      expect(description).toBeInTheDocument();
    });
  });

  describe('Responsiveness', () => {
    it('Should render avatar with correct size for small screens', () => {
      (useMediaQuery as Mock).mockReturnValue(true);
      render(<Home />);
      const avatar = screen.getByAltText('Foto de rosto do proprietário do website: Rogério Yokoi');
      expect(avatar).toHaveAttribute('width', '350');
      expect(avatar).toHaveAttribute('height', '350');
    });

    it('Should render avatar with correct size for large screens', () => {
      (useMediaQuery as Mock).mockReturnValue(false);
      render(<Home />);
      const avatar = screen.getByAltText('Foto de rosto do proprietário do website: Rogério Yokoi');
      expect(avatar).toHaveAttribute('width', '450');
      expect(avatar).toHaveAttribute('height', '450');
    });
  });
});
