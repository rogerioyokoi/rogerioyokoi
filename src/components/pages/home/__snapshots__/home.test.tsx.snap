// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components > Pages > Home > Snapshot > Should match snapshot 1`] = `
<div>
  <div
    class="flex flex-col-reverse xl:flex-row gap-12 w-100 items-center"
  >
    <div
      class="w-100 xl:w-2/4 flex justify-center"
    >
      <div
        class="relative overflow-hidden rounded-full border-amber-500 border-4 bg-gray-200"
        data-testid="avatar-container"
        style="width: 450px; height: 450px;"
      >
        <picture
          data-testid="avatar-picture"
        >
          <img
            alt="Foto de rosto do proprietário do website: Rogério Yokoi"
            class="object-cover w-full h-full opacity-0"
            data-testid="avatar-image"
            height="450"
            src="/src/assets/pessoal.jpg"
            width="450"
          />
        </picture>
        <div
          class="absolute inset-0 bg-gray-300 animate-pulse"
          data-testid="avatar-loading"
        />
      </div>
    </div>
    <div
      class="w-100 xl:w-2/4 flex flex-col gap-12"
    >
      <div
        class="uppercase text-bold"
      >
        <div
          class="flex gap-2"
        >
          <div
            aria-hidden="true"
            class="relative w-6"
          >
            <div
              class="absolute left-0 top-[44%] h-[4px] w-6 bg-amber-500 rounded-[10px]"
            />
          </div>
          <div>
            <h1
              class="text-amber-500 text-3xl lg:text-4xl"
              data-testid="title-heading"
            >
              Olá, eu sou Rogério Yokoi
            </h1>
          </div>
        </div>
        <div
          class="flex text-lg lg:text-2xl ml-8"
          data-testid="title-subheading"
        >
          Engenheiro de software Front-End
        </div>
      </div>
      <p
        data-testid="home-description"
      >
        Sou um brasileiro, situado em Minas Gerais. Um engenheiro de software front-end, motivado pela minha paixão em criar experiências digitais intuitivas e refinadas. Especializo-me em desenvolver interfaces limpas e amigáveis para usuário, assim como aplicações web. Meu compromisso está em criar soluções de software excepcionais que impactem positivamente a vida dos usuários e enriqueçam suas interações digitais.
      </p>
      <div
        class="flex flex-col sm:flex-row w-full justify-between flex-wrap gap-6"
      >
        <a
          class="flex items-center justify-between gap-4 rounded-full transition-colors ease-in-out duration-300 font-medium pl-6 text-lg bg-amber-400 hover:bg-amber-600 text-slate-900"
          data-testid="button-anchor"
          download=""
          href="/src/assets/curriculo-rogerio-yokoi.pdf"
          target="new"
        >
           
          <span
            class="w-100"
          >
            Baixar Curriculo
          </span>
           
          <div
            class="flex justify-center items-center h-12 w-12 p-3 rounded-full shadow-left bg-amber-500 text-white"
            data-testid="icon-container"
          >
            <svg
              aria-hidden="true"
              data-slot="icon"
              fill="currentColor"
              viewBox="0 0 16 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.75 2.75a.75.75 0 0 0-1.5 0v5.69L5.03 6.22a.75.75 0 0 0-1.06 1.06l3.5 3.5a.75.75 0 0 0 1.06 0l3.5-3.5a.75.75 0 0 0-1.06-1.06L8.75 8.44V2.75Z"
              />
              <path
                d="M3.5 9.75a.75.75 0 0 0-1.5 0v1.5A2.75 2.75 0 0 0 4.75 14h6.5A2.75 2.75 0 0 0 14 11.25v-1.5a.75.75 0 0 0-1.5 0v1.5c0 .69-.56 1.25-1.25 1.25h-6.5c-.69 0-1.25-.56-1.25-1.25v-1.5Z"
              />
            </svg>
          </div>
        </a>
        <a
          class="flex items-center justify-between gap-4 rounded-full transition-colors ease-in-out duration-300 font-medium pl-6 text-lg bg-blue-400 hover:bg-blue-600"
          data-testid="button-anchor"
          href="https://www.linkedin.com/in/rogerioyokoi/"
          target="new"
        >
           
          <span
            class="w-100"
          >
            Linked IN
          </span>
           
          <div
            class="flex justify-center items-center h-12 w-12 p-3 rounded-full shadow-left"
            data-testid="icon-container"
            style="background-color: rgb(0, 119, 181); color: rgb(255, 255, 255);"
          >
            <svg
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.98 3.5c0 1.381-1.11 2.5-2.48 2.5s-2.48-1.119-2.48-2.5c0-1.38 1.11-2.5 2.48-2.5s2.48 1.12 2.48 2.5zm.02 4.5h-5v16h5v-16zm7.982 0h-4.968v16h4.969v-8.399c0-4.67 6.029-5.052 6.029 0v8.399h4.988v-10.131c0-7.88-8.922-7.593-11.018-3.714v-2.155z"
              />
            </svg>
          </div>
        </a>
        <a
          class="flex items-center justify-between gap-4 rounded-full transition-colors ease-in-out duration-300 font-medium pl-6 text-lg bg-neutral-800 hover:bg-neutral-950"
          data-testid="button-anchor"
          href="https://github.com/rogerioyokoi"
          target="new"
        >
           
          <span
            class="w-100"
          >
            Github
          </span>
           
          <div
            class="flex justify-center items-center h-12 w-12 p-3 rounded-full shadow-left bg-neutral-800 dark:bg-neutral-200 dark:text-neutral-800 text-neutral-200"
            data-testid="icon-container"
          >
            <svg
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 0c-6.626 0-12 5.373-12 12 0 5.302 3.438 9.8 8.207 11.387.599.111.793-.261.793-.577v-2.234c-3.338.726-4.033-1.416-4.033-1.416-.546-1.387-1.333-1.756-1.333-1.756-1.089-.745.083-.729.083-.729 1.205.084 1.839 1.237 1.839 1.237 1.07 1.834 2.807 1.304 3.492.997.107-.775.418-1.305.762-1.604-2.665-.305-5.467-1.334-5.467-5.931 0-1.311.469-2.381 1.236-3.221-.124-.303-.535-1.524.117-3.176 0 0 1.008-.322 3.301 1.23.957-.266 1.983-.399 3.003-.404 1.02.005 2.047.138 3.006.404 2.291-1.552 3.297-1.23 3.297-1.23.653 1.653.242 2.874.118 3.176.77.84 1.235 1.911 1.235 3.221 0 4.609-2.807 5.624-5.479 5.921.43.372.823 1.102.823 2.222v3.293c0 .319.192.694.801.576 4.765-1.589 8.199-6.086 8.199-11.386 0-6.627-5.373-12-12-12z"
              />
            </svg>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
